server/download.go

```diff
@@ -451,15 +452,16 @@ func (b *blobDownload) Wait(ctx context.Context, fn func(api.ProgressResponse))
 	}
 }
 
-type downloadOpts struct {
-	mp      ModelPath
+type downloadOptions struct {
+	name    model.Name
+	baseURL *url.URL
```

Comment:
The `scheme` is a valid part of the model name, but that wasn't included in `types/model`. This would be cleaner to have something like `name.BaseURL()` which derives from the model name itself.